### PR TITLE
[FLINK-2149][gelly] Simplified Jaccard Example

### PR DESCRIPTION
**Pull request by andralungu**
This PR simplifies Gelly's Jaccard example by using the more efficient reduceOnNeighbors rather than groupReduceOnNeighbors.

### CODE REVIEW

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/JaccardSimilarityMeasure.java [hunk 1]
```diff
@@ -66,34 +63,47 @@ public static void main(String [] args) throws Exception {
...
 					@Override
-					public Double map(Tuple2<Double, Double> value) throws Exception {
-						return value.f1;
+					public HashSet<Long> map(Tuple2<HashSet<Long>, HashSet<Long>> tuple2) throws Exception {
+						return tuple2.f1;
 					}
-				}).getEdges();
+				}).getVertices();
+
+		Graph<Long, HashSet<Long>, Double> graphWithVertexValues = Graph.fromDataSet(verticesWithNeighbors, edges, env);
```
Comment (vasia) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/JaccardSimilarityMeasure.java]: joinWithVertices can give you the Graph directly :)

**Comment by vasia** [comment]
+1 except the minor comment

**Comment by andralungu** [comment]
PR updated.

**Comment by andralungu** [comment]
Meging...

**Pull request closed without merging**